### ISSUE TITLE
pihole DNS not working with custom DNS servers 

### ISSUE THREAD

**Issue by gnfpt** [pi-hole/pi-hole#2393]
**In raising this issue, I confirm the following:** `{please fill the checkboxes, e.g: [X]}`

- [X] I have read and understood the [contributors guide](https://github.com/pi-hole/pi-hole/blob/master/CONTRIBUTING.md).
- [X] The issue I am reporting can be *replicated*.
- [X] The issue I am reporting isn't a duplicate (see [FAQs](https://github.com/pi-hole/pi-hole/wiki/FAQs), [closed issues](https://github.com/pi-hole/pi-hole/issues?utf8=%E2%9C%93&q=is%3Aissue%20is%3Aclosed%20), and [open issues](https://github.com/pi-hole/pi-hole/issues)).

**How familiar are you with the the source code relevant to this issue?:*
*
1

---
**Expected behaviour:**
After setting up custom DNS (ie: 208.67.222.123 and 208.67.220.123). instead of the available DNS, shouldn't make dns resolution unavailable. 

**Actual behaviour:**
When I setup custom DNS server, within the GUI, the DNS server appears as 208.67.222.123#53 instead of 208.67.222.123 (in the configuration file it has the #53 too). Even if I remove the #53 the DNS service just doesn't work.

`{A detailed description and/or screenshots of what you do see}`

![opera_2018-08-26_22-51-28](https://user-images.githubusercontent.com/9711909/44633415-de95c500-a982-11e8-82ed-4b21702c6532.png)

![opera_2018-08-26_22-54-38](https://user-images.githubusercontent.com/9711909/44633435-24528d80-a983-11e8-9652-e001d212e007.png)

**Steps to reproduce:**

Configure the above mentioned DNS servers, for instance.

**Debug token provided by [uploading `pihole -d` log](https://discourse.pi-hole.net/t/the-pihole-command-with-examples/738#debug):**

`{Alphanumeric token}`

**Troubleshooting undertaken, and/or other relevant information:**

`{Steps of what you have done to fix this}`

**Comment by technicalpyro** [comment]
The `#53` is required as custome DNS servers allow for differentports other than the default.  Do not remove it and it will function as expected.

**Comment by gnfpt** [comment]
With or without the #53, the end result is the same: It doesn't work. I had to go with one of the Upstream DNS Servers listed.

**Comment by technicalpyro** [comment]
I cannot reproduce this myself.  i personally have Unbound running beside Pi-hole on port `5353`  Do you have anything else running on the device also what shows if you run the following command 

`dig google.com @208.67.222.123`

**Comment by gnfpt** [comment]
Wait, I forgot to unselect the cloudflare.  It's not working anymore.
If I leave openDNS it just stops working. I suspect it's the same with any other DNS server outside that list.

**Comment by gnfpt** [comment]
This Pi is dedicated to pihole. It was running fine with previous version but I had the unfortunate idea of upgrading to the last version... Also, having dnsmasq built in sounds like a bad idea: I update Raspbian more often than I update pihole.

**Comment by technicalpyro** [comment]
so this points to something deeper the command i had you run before runs dig against one of the two your are having issues with and worked perfectly.  

could you run `pihole -d` when it is not working 
then get it working in your manner 
then `cat /var/log/pihole_debug.log | pihole tricorder

**Comment by gnfpt** [comment]
I suppose uploading the logs is required...

**Comment by gnfpt** [comment]
the debug command: e19yr4vdip
the cat command: m1zeuf8yf4

**Comment by gnfpt** [comment]
What do you suggest as a starting point?

**Comment by technicalpyro** [comment]
```
addn-hosts=/etc/pihole/gravity.list
addn-hosts=/etc/pihole/black.list
addn-hosts=/etc/pihole/local.list


localise-queries


no-resolv



cache-size=10000

log-queries
log-facility=/var/log/pihole.log

local-ttl=2

log-async
server=127.0.0.1#5353
interface=eth0
```
is the content of mine 

`127.0.0.1#5353` is because i run unbound beside Pi-hole on the samer device but you can use your usual DNS servers there

**Comment by Mcat12** [comment]
`pihole -r` should also reset the config to the expected state.

**Comment by gnfpt** [comment]
Ok, I've identified the issue. 

If I setup a custom dns services with DNSSEC enabled it simply doesn't work (and before it did).
If I pick one from the list of DNS service (at least with Cloudflare) with DNSSEC enbled I have no issues.

I didn't test enough but if the dns provider must support dnssec to simply work when having dnssec enabled then either it wasn't working before or something is wrong/different now because I've always had that option enabled without prior issues.

If not working when DNS service doesn't support DNSSEC and the option is enabled is the expected result then the GUI should mention that clearly.

Thanks,
~/gnfpt

**Comment by Mcat12** [comment]
There is a note below the setting:
> Use Google, Norton, DNS.WATCH or Quad9 DNS servers when activating DNSSEC.

However, the message can be expanded to include all resolvers that support DNSSEC.

**Comment by gnfpt** [comment]
That message doesn't say that DNS won't work at all if not used with one of those DNS resolvers (or one that supports DNSSEC). 
Also,, from my previous experience with that setting, DNS continued to work if the resolver didn't support dnssec, which wasn't the case after the upgrade, DNS just stopped working while that configuration was enabled. Either it was broken before or it's broken now because I didn't change my configuration when I upgraded to the new version.

**Comment by Mcat12** [comment]
Nothing was done that would cause it to break if it worked like that before.

There is an open PR to update the wording.
pi-hole/AdminLTE#847

**Comment by gnfpt** [comment]
Well, you've included a fork of dnsmasq into pihole. Couldn't that be the cause of this?

**Comment by Mcat12** [comment]
No, none of that code has changed. The only thing that might have changed was that the version of dnsmasq used (latest) had a fix for a dnssec issue. Then it's a dnsmasq bug.

**Comment by gnfpt** [comment]
Since you’ve incorporated dnsmasq into pihole it’s really pihole problem now since fixing is done through pihole releases.

**Comment by Mcat12** [comment]
Sorry, that's not how working with upstream code bases works.
All that we have done is integrated further with dnsmasq, as if we added it as a library. We have not changed how it functions; we just have more access to its data. Whenever a new version of dnsmasq ships, we update to it. For example, dnsmasq 2.80 is nearing release, and when it does release, FTL will have all of its changes and bugfixes.

If FTL was installed separately from same version of dnsmasq, you would have the same issue.

**Comment by gnfpt** [comment]
Fixes to dnsmasq are made available through the package manager or from installing a new version of pihole? If dnsmasq was made available separately I could rely on the package manager. perhaps I could install a previous working version or leave it at a stable working version. Like I said, when you further integrate and deploy other software with your own, their problems are your problems too. 

But if you don't see it that way, good for you.

**Comment by Mcat12** [comment]
This issue (documentation on DNSSEC) is fixed with pi-hole/AdminLTE#847
